### ISSUE TITLE
Add an offline display refresh rate option (decoupled from online api polling)

### ISSUE THREAD

**Issue by lrq3000** [breezy-weather/breezy-weather#487]
### Describe your idea

Currently the automatic refresh of the main temperature (at the top of the card, in widgets and in the taskbar and lockscreen) is tied to the global refresh rate, which first polls online APIs and then refresh the various displays.

Now for most weather providers it's useless to poll more frequently than about every 6h because their models do not output updated predictions more frequently than that (and for some it's even more infrequent with a refresh only at midnight everyday). So to be more ecologically responsible, it's possible to set a refresh rate of 6h.

The problem is that then all displays are only refreshed once per 6h, so at noon i can still have the temperature at 6am displayed.

I would like to suggest to add another option to set a separate display refresh rate to allow for a more frequent display refresh rate without the online API polling, so that I could set the app to refresh the display every 30min for example but only poll once every 6h, which would be a very ecological solution to reduce online polling rate while just incurring a negligible impact on user's battery if chosen.

This would require decoupling the code for online polling and refreshing display, bu i believe this has a big enough ecological impact to be considered as an interesting suggestion.

### Acknowledgements

- [X] I have searched the existing issues and this is a new ticket, **NOT** a duplicate or related to another open or closed issue.
- [X] I have written a short but informative title.
- [X] My idea is not related to a question covered in the [Frequently Asked Questions](https://github.com/breezy-weather/breezy-weather/blob/main/HELP.md) document, or a feature described in the [Homepage](https://github.com/breezy-weather/breezy-weather/blob/main/docs/HOMEPAGE.md) document.
- [X] I have updated the app to **[version 4.6.7 or later](https://github.com/breezy-weather/breezy-weather/releases/latest)**.
- [X] I understand that this app is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY, so I may not get a positive reply to my request, if any at all.

**Comment by papjul** [comment]
I didn't think of this specific issue but I had a similar need for different refresh times by locations, or for spreading refreshes accross the day (especially if you refreshed manually in-between, it will keep updating all locations unless it's been less than X minutes (15 I think?)).

The only way I could think of was to set up a periodic job every 30 min (matching the minimal refresh periodic), not so green friendly. In your case, I should additionally remove the network constraint, making it less optimized to resume as soon as network is available.

Or, in your specific case, I could have a different periodic run, but what would be a good delay between each? Are you aware some manufacturers don't like apps that set up too many jobs or with too short delays, and just kill them without any notice?

So I just gave up thinking about this, but I think the two issues should be thought at the same time.

Another thing to solve is current. You say forecast models are only refreshed every 6 hours, but that's for long-term forecasts. That's not the case for 1 or 15 min forecast, or alert updates (hence why the caching policy on these ones are more light). Even so, you want your widgets to update, but we can only show the current from the previous refresh, which would still show the same time as previous refresh because that's when current was valid. Example: you refresh at 06:00, you update widgets without network at 09:00 => widgets still display 06:00 because that's the date of validity.

**Comment by lrq3000** [comment]
I'm not sure I was clear enough with my description, or even that my suggested solution was adequate on second thought.

The issue I am reporting here about is that the values displayed by the app, whether on widgets or at the top of cards, reflect the data at the time of the last sync, instead of current.

Let's take an example: I synced at 6am. Now it's 10am, but I did not sync. The app already has predictions for 10am, so I expected it to show the temperature it predicts at 10am. But in fact it still shows the temperature at 6am.

I argue this is not an expected behavior. Furthermore, even if the most "reliable" temperature data point is the one at sync, it still represents a prediction and one that can be remote from ground truth.

Instead, the app should show the value corresponding to the current time at any time, regardless of last sync, it should fetch in its already stored predictions. There is no need for a different display refresh timer in app, except maybe for  widgets (I don't know how they work - do they poll your app or you have to refresh them explicitly in the code?).

In summary, the issue I report about is that I argue that tying the display to the last sync time is counter-intuitive for most users, it should at the very least be an option to enable/disable.

**Comment by papjul** [comment]
> Let's take an example: I synced at 6am. Now it's 10am, but I did not sync. The app already has predictions for 10am, so I expected it to show the temperature it predicts at 10am. But in fact it still shows the temperature at 6am.

I understood your idea, but the values you see are observations (current), not predictions (hourly forecast). There is no way to get the updated observations from 10:00 if you only synced at 06:00. Observations are only from the exact moment you synced. So it will always show the observations from 06:00 until your next sync where you will get updated observations.

**Comment by lrq3000** [comment]
Are you sure they are observations? I remember seeing big discrepancies between what was "observed" by the weather sources (in France) versus what was observable in reality. Do you have documentation about this for at least one weather forecast provider so that I can lookup this type of data, as I did not know they provided observations on top of forecasts.

In case these are observations as you say, then my points above are moot and this issue can be closed, thank you for your consideration.

**Comment by papjul** [comment]
Météo-France has 1150 observation stations from the RADOME network in France.

Example of station:
https://www.meteociel.fr/temps-reel/obs_villes.php?code2=7560&affint=1
(You can tap on "Fiche station" for more info)

Observations are updated every 6 min, and you can indeed compare the values in the app; they are the same with a 6-min precision.

There are sources without observation stations, or only for specific countries, in that case they will usually fallback to something else, but they will do it by themself.

For example, Open-Meteo "current" will fallback to closest 15-min forecast (which we don't have in-app, we only have hourly forecast), but it's all transparent for us, Open-Meteo takes care of that by itself.

**Comment by lrq3000** [comment]
Also I'd like to mention that if I need a more up-to-date forecast right now, I can manually refresh. My point is that most of the time we don't need so much precision, we need only once in a while. Another solution would be to set the syncing to be totally manual, but this is unsatisfactory for most because for mundane purposes we still need forecasts at least daily or every few days, which forces the user set to manual to always manually refresh each day.

So that's why I thought of a greener approach being one daily internet syncing + manual refresh if necessary, but with display updated to reflect the forecast of current time, not the last observation/forecast at synced time.

**Comment by lrq3000** [comment]
It seems we are not understanding each other and things appear to drift into arguing instead of constructing. I think it's better to just drop my request. Thank you for your consideration.
